### ISSUE TITLE
Kernel not being set on notebook load

### ISSUE THREAD

**Issue by jtyberg** [ipython/ipython#7645]
The notebook kernel is not being set on notebook load, which can cause some confusion. 

I pulled 3.0.0b1 and immediately uploaded an existing notebook.  The first cell is `!pip install redis`, which ran fine, but my subsequent `import redis` failed.  I sat frozen for a bit before it hit me that in IPy3 I have to think about which kernel I'm using.  I'm not sure if new/uploaded notebooks default to a Python 3 kernel, as the indicator just said, 'Kernel'.

Long story short, I had to make sure my kernel selection matched my pip/pip3 install to get it to work properly.

![kernel_not_set](https://cloud.githubusercontent.com/assets/359081/5986545/464ed050-a8ca-11e4-9373-710a58a29dec.gif)

**Comment by takluyver** [comment]
Specifically, it looks like the kernel indicator is not being updated. I see an error message in there 'Cannot change kernel while waiting for pending session start' - I bet that's related.

**Comment by minrk** [comment]
This was almost certainly caused by my changes to the kernelspec loading events, and I saw it yesterday as well. I'll try to track it down.

**Comment by minrk** [comment]
Should be fixed by #7650
